packages/neon_framework/packages/neon_http_client/lib/src/interceptors/authentication_interceptor.dart

```diff
@@ -0,0 +1,37 @@
+import 'package:http/http.dart';
+import 'package:meta/meta.dart';
+import 'package:neon_http_client/src/interceptors/http_interceptor.dart';
+import 'package:nextcloud/nextcloud.dart';
+
+@internal
+final class AuthenticationInterceptor implements HttpInterceptor {
+  var _blocked = false;
+
+  @override
+  bool shouldInterceptRequest(BaseRequest request) {
+    final authorization = request.headers['authorization'];
+    return authorization != null && authorization.isNotEmpty && _blocked;
+  }
+
+  @override
+  Never interceptRequest({required BaseRequest request}) {
+    throw DynamiteStatusCodeException(Response('', 401));
+  }
+
+  @override
+  bool shouldInterceptResponse(StreamedResponse response) {
+    final authorization = response.request?.headers['authorization'];
+    if (authorization == null || authorization.isEmpty) {
+      return false;
+    }
+
+    return response.statusCode == 401 || (response.request?.url.path.endsWith('/index.php/login/v2/poll') ?? false);
```

Comment:
1. Please add the baseURL as a parameter to this class so you can match the real URL.
2. please move the url matching to the interception method. The url there might be more accurate.